include/slam_toolbox/experimental/slam_toolbox_lifelong.hpp

```diff
@@ -48,6 +50,29 @@ class LifelongSlamToolbox : public SlamToolbox
     LocalizedRangeScan * s1, LocalizedRangeScan * s2, double & x_l,
     double & x_u, double & y_l, double & y_u);
 
+public:
+  // Cell occupancy struct for unordered map
+  struct Occupancy
+  {
+    int fr, oc ,un;
+    bool operator==(Occupancy const& st) const
+    {
+      return (st.fr == fr) && (st.oc == oc) && (st.un == un);
+    }
+    
+    struct CombinationsHash
+    {
+      std::size_t operator()(Occupancy const& key) const 
+      {
+        std::size_t hash = 5381u;
+        hash = (hash << 5) + hash + key.fr;
+        hash = (hash << 5) + hash + key.oc;
+        hash = (hash << 5) + hash + key.un;
+        return hash;
+      }
+    };
+  };
```

Comment:
@kmilo7204 have you considered using an [`std::tuple`](https://en.cppreference.com/w/cpp/utility/tuple)?

Reply:
@hidmic, thanks for the suggestion. Just a quick question:

Should we use the std::tuple for replacing the whole structure (Occupancy), or for replacing the structure members (fr, oc, un)?

Reply:
@kmilo7204 the whole `struct` (which _is_ conceptually a tuple).